### PR TITLE
saw-shortening a doublebarrel shotgun now turns it into a subtype

### LINKED ISSUE #28187: double barrel shotgun sprite error
(WEB REPORT BY: generalapathy REMOTE: 206.221.180.138:7777)
# Revision
c2b50c04da7af68785d08a79730e1ac0eab58fd2

# Description
when you make the bar shotgun a sawn off, and then reload it, the sprite becomes the regular model. 
# Steps to Reproduce
saw off the gun with any saw and then load the shells, i only tried it with beanbag shells so that might be it
# What you Expected
the sprite not to change
# What Actually Happened
the shotgun barrel grew three inches that day.

### PR DESCRIPTION
**Pull request by SonixApache**
this fixes #28187

### CODE REVIEW

File: code/modules/projectiles/guns/projectile/shotgun.dm [hunk 1]
```diff
@@ -149,17 +149,18 @@
...
 			return
 		if(do_after(user, src, 30))	//SHIT IS STEALTHY EYYYYY
-			icon_state = "sawnshotgun"
-			w_class = W_CLASS_MEDIUM
-			item_state = "sawnshotgun"
-			slot_flags &= ~SLOT_BACK	//you can't sling it on your back
-			slot_flags |= SLOT_BELT		//but you can wear it on your belt (poorly concealed under a trenchcoat, ideally)
-			name = "sawn-off shotgun"
-			desc = "Omar's coming!"
+			var/obj/item/weapon/gun/projectile/shotgun/doublebarrel/sawnoff/itssmallnow = new /obj/item/weapon/gun/projectile/shotgun/doublebarrel/sawnoff(src.loc)
```
Comment (SonixApache) [inline code/modules/projectiles/guns/projectile/shotgun.dm]: let me know if the `(src.loc)` on this line is 🆗

File: code/modules/projectiles/guns/projectile/shotgun.dm [hunk 2]
```diff
@@ -149,17 +149,18 @@
...
+			for(var/obj/item/ammo_casing/shotgun/loaded_shell in itssmallnow)
+				qdel(loaded_shell)
+			itssmallnow.loaded = null
 			to_chat(user, "<span class='warning'>You shorten the barrel of \the [src]!</span>")
 			if(istype(user, /mob/living/carbon/human) && src.loc == user)
 				var/mob/living/carbon/human/H = user
+				H.drop_item(src, force_drop = 1)
+				H.put_in_hands(itssmallnow)
 				H.update_inv_hands()
+				src.transfer_fingerprints_to(itssmallnow)
```
Comment (DamianX) [inline code/modules/projectiles/guns/projectile/shotgun.dm]: why are you only transferring fingerprints if the shotgun was in the user's hand?

File: code/modules/projectiles/guns/projectile/shotgun.dm [hunk 3]
```diff
@@ -149,17 +149,18 @@
...
 		if(do_after(user, src, 30))	//SHIT IS STEALTHY EYYYYY
-			icon_state = "sawnshotgun"
-			w_class = W_CLASS_MEDIUM
-			item_state = "sawnshotgun"
-			slot_flags &= ~SLOT_BACK	//you can't sling it on your back
-			slot_flags |= SLOT_BELT		//but you can wear it on your belt (poorly concealed under a trenchcoat, ideally)
-			name = "sawn-off shotgun"
-			desc = "Omar's coming!"
+			var/obj/item/weapon/gun/projectile/shotgun/doublebarrel/sawnoff/itssmallnow = new /obj/item/weapon/gun/projectile/shotgun/doublebarrel/sawnoff(src.loc)
+			for(var/obj/item/ammo_casing/shotgun/loaded_shell in itssmallnow)
```
Comment (DamianX) [inline code/modules/projectiles/guns/projectile/shotgun.dm]: you could make a subtype that starts with no shells in the first place instead

File: code/modules/projectiles/guns/projectile/shotgun.dm (hunk 2 above)
Comment (SonixApache) [reply code/modules/projectiles/guns/projectile/shotgun.dm]: i am dumb

File: code/modules/projectiles/guns/projectile/shotgun.dm (hunk 3 above)
Comment (SonixApache) [reply code/modules/projectiles/guns/projectile/shotgun.dm]: i've had bad experiences with such things in the past but that'd probably work
is it necessary, though?

Comment (DamianX) [reply code/modules/projectiles/guns/projectile/shotgun.dm]: It's not a big deal either way, but I don't like creating things (the bullets in the shotgun's New) only to delete them immediately after. It's not free, but it's also not expensive.

Comment (SonixApache) [reply code/modules/projectiles/guns/projectile/shotgun.dm]: makes sense

**Pull request merged**